platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/status/PlatformStatusStateMachine.java

```diff
@@ -16,166 +16,28 @@
-        this.currentStatusLogic = new StartingUpStatusLogic(config);
-        this.currentStatusStartTime = time.now();
-    }
-
-    /**
-     * Passes the received action into the logic method that corresponds with the action type, and returns whatever that
-     * logic method returns.
-     * <p>
-     * If the logic method throws an {@link IllegalPlatformStatusException}, this method will log the exception and
-     * return null.
-     *
-     * @param action the action to process
-     * @return a new logic object, or null if the logic method threw an exception
-     */
-    @Nullable
-    private PlatformStatusLogic getNewLogic(@NonNull final PlatformStatusAction action) {
-        Objects.requireNonNull(action);
-
-        final Class<? extends PlatformStatusAction> actionClass = action.getClass();
-
-        try {
-            if (actionClass == CatastrophicFailureAction.class) {
-                return currentStatusLogic.processCatastrophicFailureAction((CatastrophicFailureAction) action);
-            } else if (actionClass == DoneReplayingEventsAction.class) {
-                return currentStatusLogic.processDoneReplayingEventsAction((DoneReplayingEventsAction) action);
-            } else if (actionClass == FallenBehindAction.class) {
-                return currentStatusLogic.processFallenBehindAction((FallenBehindAction) action);
-            } else if (actionClass == FreezePeriodEnteredAction.class) {
-                return currentStatusLogic.processFreezePeriodEnteredAction((FreezePeriodEnteredAction) action);
-            } else if (actionClass == ReconnectCompleteAction.class) {
-                return currentStatusLogic.processReconnectCompleteAction((ReconnectCompleteAction) action);
-            } else if (actionClass == SelfEventReachedConsensusAction.class) {
-                return currentStatusLogic.processSelfEventReachedConsensusAction(
-                        (SelfEventReachedConsensusAction) action);
-            } else if (actionClass == StartedReplayingEventsAction.class) {
-                return currentStatusLogic.processStartedReplayingEventsAction((StartedReplayingEventsAction) action);
-            } else if (actionClass == StateWrittenToDiskAction.class) {
-                return currentStatusLogic.processStateWrittenToDiskAction((StateWrittenToDiskAction) action);
-            } else if (actionClass == TimeElapsedAction.class) {
-                return currentStatusLogic.processTimeElapsedAction((TimeElapsedAction) action);
-            } else {
-                throw new IllegalArgumentException(
-                        "Unknown action type: " + action.getClass().getName());
-            }
-        } catch (final IllegalPlatformStatusException e) {
-            logger.error(EXCEPTION.getMarker(), e.getMessage(), e);
-            return null;
-        }
-    }
-
+public interface PlatformStatusStateMachine extends PlatformStatusGetter, Startable, Stoppable {
```

Comment:
It doesn't make sense to me to have the interface extend `Startable` and `Stoppable` when they only apply to one of the two implementations. I'm not sure this interface should even exist. There is one class that contains the queue thread: `AsyncPlatformStatusStateMachine`. I think it should instantiate the `SyncPlatformStatusStateMachine` class inside its own constructor and use it as a helper class that is strictly internal to itself, in which case the interface is not needed.

In my view, the `AsyncPlatformStatusStateMachine` is the "task component" interface that the outside world interacts with. That task module instantiates classes it needs to do it's job, one of which is `SyncPlatformStatusStateMachine`. Looking at it this way, I think it is more clear to call these class something like `PlatformStatusComponent` and `PlatformStatusStateMachine`.

Reply:
I agree. Reworked

Reply:
Instead of `PlatformStatusComponent`, how about the name `PlatformStatusManager`? Even if this class is using Lazar's new component architecture, "component" seems linguistically awkward to me in this context.

Reply:
I also prefer `PlatformStatusManager`

Reply:
Renamed